### PR TITLE
refactor method change_cell_style in zcl_excel_worksheet

### PR DESCRIPTION
**Pull request by sandraros**
Creation of new class `zcl_excel_style_changer` and refactor of `change_cell_style` to use it. There's no fix, no feature change, it helps to go towards #524.

Future plan: with this new class, it will be easy to create a new method `change_area_style` so that to change the style of many cells at a time. That would permit to rewrite [`zdemo_excel40`](https://github.com/sapmentors/abap2xlsx/blob/master/src/demos/zdemo_excel40.prog.abap), for the exact same result, instead of calling `set_cell` and `change_cell_style` repeatedly for each cell in 100 rows * 20 columns, the methods `set_area` and `change_area_style` could be called once:
```abap
    lo_worksheet->set_area(
          ip_column_start = 1
          ip_column_end   = 20
          ip_row          = 1
          ip_row_to       = 100
          ip_formula      = 'CHAR(64+COLUMN())&TEXT(ROW(),"       ?0")' ).

    lo_changer = zcl_excel_style_changer=>create( lo_excel ).
    lo_changer->set_fill_filltype( zcl_excel_style_fill=>c_fill_solid ).
    lo_changer->set_fill_fgcolor_rgb( zcl_excel_style_color=>c_yellow ).
    lo_worksheet->change_area_style(
          ip_column_start = 1
          ip_column_end   = 20
          ip_row          = 1
          ip_row_to       = 3
          ip_changer      = lo_changer ).

    lo_changer = zcl_excel_style_changer=>create( lo_excel ).
    lo_changer->set_font_color_rgb( zcl_excel_style_color=>c_red ).
    lo_worksheet->change_area_style(
          ip_column_start = 1
          ip_column_end   = 4
          ip_row          = 1
          ip_row_to       = 100
          ip_changer      = lo_changer ).
```

### CODE REVIEW

File: src/zcl_excel_style_changer.clas.abap [hunk 1]
```diff
@@ -0,0 +1,1718 @@
...
+    DATA: style TYPE REF TO zcl_excel_style_changer.
+
+    CREATE OBJECT style.
+    style->excel = excel.
+    result = style.
+
+*  ENDMETHOD.
+*
+*
+*  METHOD get.
```
Comment (AndreaBorgia-Abo) [inline src/zcl_excel_style_changer.clas.abap]: I'm not sure I understand the reason behind these two methods, commented out and left there.

File: src/zcl_excel_worksheet.clas.abap [hunk 2]
```diff
@@ -1387,619 +1387,374 @@ CLASS zcl_excel_worksheet IMPLEMENTATION.
...
-      complete_style-borders-top-border_color-tint = ip_borders_top_color_tint.
-      complete_stylex-borders-top-border_color-tint = 'X'.
+      changer->set_borders_top_color_tint( ip_borders_top_color_tint ).
     ENDIF.
 
 
-* Now we have a completly filled styles.
-* This can be used to get the guid
-* Return guid if requested.  Might be used if copy&paste of styles is requested
-    ep_guid = me->excel->get_static_cellstyle_guid( ip_cstyle_complete  = complete_style
```
Comment (AndreaBorgia-Abo) [inline src/zcl_excel_worksheet.clas.abap:1999]: Where is ep_guid filled in in the refactored code? Debugged twice (new / old) and it seems te be missing.

File: src/zcl_excel_style_changer.clas.abap [hunk 3]
```diff
@@ -0,0 +1,1718 @@
...
+
+    CLASS-METHODS create
+      IMPORTING
+        excel         TYPE REF TO zcl_excel
+      RETURNING
+        VALUE(result) TYPE REF TO zif_excel_style_changer
+      RAISING
+        zcx_excel.
+
+*    CLASS-METHODS get
```
Comment (AndreaBorgia-Abo) [inline src/zcl_excel_style_changer.clas.abap]: I'm not sure I understand the reason behind these two methods, commented out and left there.

**Review by AndreaBorgia-Abo — CHANGES_REQUESTED** [review]
There seems to be a regression with an export parameter and a couple of commented-out methods I cannot understand.

File: src/zcl_excel_worksheet.clas.abap (hunk 2 above)
Comment (sandraros) [reply src/zcl_excel_worksheet.clas.abap:1999]: You're right, regression, let me fix that.

File: src/zcl_excel_style_changer.clas.abap (hunk 1 above)
Comment (sandraros) [reply src/zcl_excel_style_changer.clas.abap]: No reason, let me remove them.

Comment (sandraros) [reply src/zcl_excel_style_changer.clas.abap]: Done

File: src/zcl_excel_worksheet.clas.abap (hunk 2 above)
Comment (sandraros) [reply src/zcl_excel_worksheet.clas.abap:1999]: Done

**Comment by sandraros** [comment]
Program for testing the `ep_guid` parameter returned by `change_cell_style`:
```abap
REPORT.
DATA: lo_excel     TYPE REF TO zcl_excel,
      lo_worksheet TYPE REF TO zcl_excel_worksheet.
CONSTANTS: gc_save_file_name TYPE string VALUE 'PR_904.xlsx'.
INCLUDE zdemo_excel_outputopt_incl.
START-OF-SELECTION.
  CREATE OBJECT lo_excel.
  lo_worksheet = lo_excel->get_active_worksheet( ).
  lo_worksheet->set_cell( ip_column = 1 ip_row = 1 ip_value = 'A1' ).
  lo_worksheet->set_cell( ip_column = 2 ip_row = 1 ip_value = 'B1' ).
  lo_worksheet->set_cell( ip_column = 3 ip_row = 1 ip_value = 'C1' ).
  DATA(lv_guid) = lo_worksheet->change_cell_style(
      ip_column           = 1
      ip_row              = 1
      ip_fill_filltype    = zcl_excel_style_fill=>c_fill_solid
      ip_fill_fgcolor_rgb = zcl_excel_style_color=>c_yellow ).
  lo_worksheet->set_cell_style( ip_column = 2 ip_row = 1 ip_style = lv_guid ).
  DATA(lv_guid2) = lo_worksheet->change_cell_style(
      ip_column         = 1
      ip_row            = 1
      ip_font_color_rgb = zcl_excel_style_color=>c_red ).
  lo_worksheet->set_cell_style( ip_column = 3 ip_row = 1 ip_style = lv_guid2 ).
  lcl_output=>output( lo_excel ).
```

**Review by AndreaBorgia-Abo — APPROVED** [review]
LGTM

**Pull request merged**